### ISSUE TITLE
Email address not validated in check()

### ISSUE THREAD

**Issue by VincentGuyader** [r-hub/rhub#45]
Hi,

My mail seems to be validated:

```
> list_validated_emails()

              email                            token
1 vincent@thinkr.fr 1bfd86e8fa0e4a5d9506e40XXXXXXX
```

But it doesn't work :(

```
> check_for_cran()
─  Building package
─  Uploading package
Error in query("SUBMIT PACKAGE", data = list(email = unbox(email), token = unbox(email_get_token(email)),  : 
  Email address not validated
```

Regards

**Comment by VincentGuyader** [comment]
everything ok on a fresh new installation...

**Comment by gaborcsardi** [comment]
I think you probably requested another token from r-hub, but you did not update the token in R. 

Try to re-validate it, make sure that you have the latest token, and try again.

Let me know if you have any problems.

**Comment by lcolladotor** [comment]
Something similar happened to me because I copied the newline character when copying the token. Re-validating it and using the latest token worked.

**Comment by gaborcsardi** [comment]
Good to know. I guess I should check that the token looks like a token.

**Comment by benmarwick** [comment]
I'm getting a similar error:

```
> rhub::list_validated_emails()
                 email                            token
1 benmarwick@gmail.com 3ceec23528c34518a247ba49585a4240
> rhub::check_for_cran()

  ‘yourself@somewhere.net’ is not validated. To validate it
  now, please enter the email address below. Note that r-hub
  will send a token to this address. If the address does not
  belong to you, quit now by pressing ENTER .

  Email address: benmarwick@gmail.com

Error: Emails don't match, aborting
``` 

I've installed the r-hub pkg from GH, then validated the email, then try to run `rhub::check_for_cran()` but it seems like the email validation hasn't succeeded. 

```
> devtools::session_info()
Session info -------------------------------------------------------------------------------------------------
 setting  value                       
 version  R version 3.3.1 (2016-06-21)
 system   x86_64, mingw32             
 ui       RStudio (1.0.5)             
 language (EN)                        
 collate  English_Australia.1252      
 tz       America/Los_Angeles         
 date     2016-12-08                  

Packages -----------------------------------------------------------------------------------------------------
 package     * version    date       source                             
 archdata    * 1.1        2016-04-05 CRAN (R 3.3.1)                     
 assertthat    0.1        2013-12-06 CRAN (R 3.2.2)                     
 beeswarm      0.2.3      2016-04-25 CRAN (R 3.3.1)                     
 callr         1.0.0.9000 2016-12-09 Github (MangoTheCat/callr@3007ba2) 
 clisymbols    1.0.0      2015-06-08 CRAN (R 3.3.0)                     
 colorspace    1.3-1      2016-11-18 CRAN (R 3.3.2)                     
 crayon        1.3.2      2016-06-28 CRAN (R 3.3.1)                     
 curl          2.3        2016-11-24 CRAN (R 3.3.2)                     
 cvequality  * 0.1.0      2016-12-09 local                              
 DBI           0.5-1      2016-09-10 CRAN (R 3.3.1)                     
 desc          1.0.1      2016-09-23 CRAN (R 3.3.1)                     
 devtools    * 1.12.0     2016-06-24 CRAN (R 3.3.1)                     
 digest        0.6.10     2016-08-02 CRAN (R 3.3.1)                     
 dplyr       * 0.5.0.9000 2016-11-07 Github (hadley/dplyr@e8083d4)      
 ggbeeswarm  * 0.5.3      2016-12-01 CRAN (R 3.3.2)                     
 ggplot2     * 2.2.0      2016-11-11 CRAN (R 3.3.2)                     
 git2r         0.16.0     2016-11-20 CRAN (R 3.3.2)                     
 gtable        0.2.0      2016-02-26 CRAN (R 3.2.4)                     
 httr          1.2.1      2016-07-03 CRAN (R 3.3.1)                     
 jsonlite      1.1        2016-09-14 CRAN (R 3.3.1)                     
 knitr       * 1.15.1     2016-11-22 CRAN (R 3.3.2)                     
 lazyeval      0.2.0.9000 2016-11-07 Github (hadley/lazyeval@c155c3d)   
 lego        * 0.2        2016-12-08 Github (seankross/lego@e71d780)    
 magrittr      1.5        2014-11-22 CRAN (R 3.3.1)                     
 memoise       1.0.0      2016-01-29 CRAN (R 3.2.5)                     
 munsell       0.4.3      2016-02-13 CRAN (R 3.2.4)                     
 parsedate     1.1.1      2014-09-24 CRAN (R 3.3.1)                     
 plyr          1.8.4      2016-06-08 CRAN (R 3.3.1)                     
 prettyunits   1.0.2      2015-07-13 CRAN (R 3.3.1)                     
 purrr       * 0.2.2      2016-06-18 CRAN (R 3.3.1)                     
 R6            2.2.0      2016-10-05 CRAN (R 3.3.1)                     
 rappdirs      0.3.1      2016-03-28 CRAN (R 3.3.1)                     
 rcmdcheck     1.2.1      2016-09-28 CRAN (R 3.3.1)                     
 Rcpp          0.12.8     2016-11-17 CRAN (R 3.3.2)                     
 readr       * 1.0.0      2016-08-03 CRAN (R 3.3.1)                     
 rematch       1.0.1      2016-04-21 CRAN (R 3.3.1)                     
 rhub        * 1.0.1      2016-12-09 Github (r-hub/rhub@60c8b6d)        
 rstudioapi    0.6        2016-07-12 Github (rstudio/rstudioapi@55d3200)
 scales        0.4.1      2016-11-09 CRAN (R 3.3.2)                     
 tibble      * 1.2        2016-08-26 CRAN (R 3.3.1)                     
 tidyr       * 0.6.0      2016-08-12 CRAN (R 3.3.1)                     
 tidyverse   * 1.0.0      2016-09-09 CRAN (R 3.3.1)                     
 vipor         0.4.4      2016-10-05 CRAN (R 3.3.1)                     
 whoami        1.1.1      2015-07-13 CRAN (R 3.3.0)                     
 withr         1.0.2      2016-06-20 CRAN (R 3.3.0) 
```

**Comment by gaborcsardi** [comment]
>  ‘yourself@somewhere.net’ is not validated. To validate it
  now, please enter the email address below. Note that r-hub
  will send a token to this address. If the address does not
  belong to you, quit now by pressing ENTER .

This information is taken from the package you are working with. You need to add a proper maintainer address to it, or supply the `email` option to `check()`.

**Comment by benmarwick** [comment]
Thanks for your quick reply. Your suggestion was perfect, I updated the maintainer email and it worked as expected. Thanks for your great work on this wonderful project. I've made a PR https://github.com/r-hub/rhub/pull/59 with a suggestion about updating the message text to inform the user that the error might be related to a bad maintainer email address, in addition to current message about external email validation. That might help others who get stuck like me!
